tests/tools/docker_memory_check_test.py

```diff
@@ -0,0 +1,106 @@
+import os
```

Comment:
It seems I can't run this test without having to initialize dev services. Would it be possible to be fixed? If it gets complicated we can put it in the backlog.

```
pytest tests/tools/docker_memory_check_test.py
/Users/armenzg/code/sentry/.venv/lib/python3.8/site-packages/django/conf/__init__.py:181: RemovedInDjango31Warning: The FILE_CHARSET setting is deprecated. Starting with Django 3.1, all files read from disk must be UTF-8 encoded.
  warnings.warn(FILE_CHARSET_DEPRECATED_MSG, RemovedInDjango31Warning)
/Users/armenzg/code/sentry/src/sentry/runner/initializer.py:187: DeprecatedSettingWarning: The GITHUB_APP_ID setting is deprecated. Please use SENTRY_OPTIONS['github-login.client-id'] instead.
  warnings.warn(DeprecatedSettingWarning(options_mapper[k], "SENTRY_OPTIONS['%s']" % k))
/Users/armenzg/code/sentry/src/sentry/runner/initializer.py:187: DeprecatedSettingWarning: The GITHUB_API_SECRET setting is deprecated. Please use SENTRY_OPTIONS['github-login.client-secret'] instead.
  warnings.warn(DeprecatedSettingWarning(options_mapper[k], "SENTRY_OPTIONS['%s']" % k))
14:06:08 [WARNING] sentry.utils.geo: settings.GEOIP_PATH_MMDB not configured.
14:06:09 [INFO] sentry.plugins.github: apps-not-configured
INTERNALERROR> Traceback (most recent call last):
INTERNALERROR>   File "/Users/armenzg/code/sentry/.venv/lib/python3.8/site-packages/redis/connection.py", line 552, in connect
INTERNALERROR>     sock = self._connect()
INTERNALERROR>   File "/Users/armenzg/code/sentry/.venv/lib/python3.8/site-packages/redis/connection.py", line 609, in _connect
INTERNALERROR>     raise err
INTERNALERROR>   File "/Users/armenzg/code/sentry/.venv/lib/python3.8/site-packages/redis/connection.py", line 597, in _connect
INTERNALERROR>     sock.connect(socket_address)
INTERNALERROR> ConnectionRefusedError: [Errno 61] Connection refused
INTERNALERROR>
INTERNALERROR> During handling of the above exception, another exception occurred:
INTERNALERROR>
INTERNALERROR> Traceback (most recent call last):
INTERNALERROR>   File "/Users/armenzg/code/sentry/src/sentry/ratelimits/redis.py", line 67, in validate
INTERNALERROR>     self.client.ping()
INTERNALERROR>   File "/Users/armenzg/code/sentry/.venv/lib/python3.8/site-packages/redis/client.py", line 1351, in ping
INTERNALERROR>     return self.execute_command('PING')
INTERNALERROR>   File "/Users/armenzg/code/sentry/.venv/lib/python3.8/site-packages/sentry_sdk/integrations/redis.py", line 101, in sentry_patched_execute_command
INTERNALERROR>     return old_execute_command(self, name, *args, **kwargs)
INTERNALERROR>   File "/Users/armenzg/code/sentry/.venv/lib/python3.8/site-packages/redis/client.py", line 875, in execute_command
INTERNALERROR>     conn = self.connection or pool.get_connection(command_name, **options)
INTERNALERROR>   File "/Users/armenzg/code/sentry/.venv/lib/python3.8/site-packages/redis/connection.py", line 1185, in get_connection
INTERNALERROR>     connection.connect()
INTERNALERROR>   File "/Users/armenzg/code/sentry/.venv/lib/python3.8/site-packages/redis/connection.py", line 557, in connect
INTERNALERROR>     raise ConnectionError(self._error_message(e))
INTERNALERROR> redis.exceptions.ConnectionError: Error 61 connecting to localhost:6379. Connection refused.
INTERNALERROR>
INTERNALERROR> During handling of the above exception, another exception occurred:
INTERNALERROR>
INTERNALERROR> Traceback (most recent call last):
INTERNALERROR>   File "/Users/armenzg/code/sentry/.venv/lib/python3.8/site-packages/_pytest/main.py", line 253, in wrap_session
INTERNALERROR>     config._do_configure()
INTERNALERROR>   File "/Users/armenzg/code/sentry/.venv/lib/python3.8/site-packages/_pytest/config/__init__.py", line 982, in _do_configure
INTERNALERROR>     self.hook.pytest_configure.call_historic(kwargs=dict(config=self))
INTERNALERROR>   File "/Users/armenzg/code/sentry/.venv/lib/python3.8/site-packages/pluggy/hooks.py", line 308, in call_historic
INTERNALERROR>     res = self._hookexec(self, self.get_hookimpls(), kwargs)
INTERNALERROR>   File "/Users/armenzg/code/sentry/.venv/lib/python3.8/site-packages/pluggy/manager.py", line 93, in _hookexec
INTERNALERROR>     return self._inner_hookexec(hook, methods, kwargs)
INTERNALERROR>   File "/Users/armenzg/code/sentry/.venv/lib/python3.8/site-packages/pluggy/manager.py", line 84, in <lambda>
INTERNALERROR>     self._inner_hookexec = lambda hook, methods, kwargs: hook.multicall(
INTERNALERROR>   File "/Users/armenzg/code/sentry/.venv/lib/python3.8/site-packages/pluggy/callers.py", line 208, in _multicall
INTERNALERROR>     return outcome.get_result()
INTERNALERROR>   File "/Users/armenzg/code/sentry/.venv/lib/python3.8/site-packages/pluggy/callers.py", line 80, in get_result
INTERNALERROR>     raise ex[1].with_traceback(ex[2])
INTERNALERROR>   File "/Users/armenzg/code/sentry/.venv/lib/python3.8/site-packages/pluggy/callers.py", line 187, in _multicall
INTERNALERROR>     res = hook_impl.function(*args)
INTERNALERROR>   File "/Users/armenzg/code/sentry/src/sentry/utils/pytest/sentry.py", line 190, in pytest_configure
INTERNALERROR>     initialize_app({"settings": settings, "options": None})
INTERNALERROR>   File "/Users/armenzg/code/sentry/src/sentry/runner/initializer.py", line 371, in initialize_app
INTERNALERROR>     setup_services(validate=not skip_service_validation)
INTERNALERROR>   File "/Users/armenzg/code/sentry/src/sentry/runner/initializer.py", line 415, in setup_services
INTERNALERROR>     service.validate()
INTERNALERROR>   File "/Users/armenzg/code/sentry/src/sentry/utils/services.py", line 127, in <lambda>
INTERNALERROR>     context[key] = (lambda f: lambda *a, **k: getattr(self, f)(*a, **k))(key)
INTERNALERROR>   File "/Users/armenzg/code/sentry/src/sentry/ratelimits/redis.py", line 69, in validate
INTERNALERROR>     raise InvalidConfiguration(str(e))
INTERNALERROR> sentry.exceptions.InvalidConfiguration: Error 61 connecting to localhost:6379. Connection refused.
```

Reply:
I added a `make test-tools` which does the necessary things here.  I would love for it to not need the `-c /dev/null --confcutdir tests/tools` options and for our `conftest` to be "more normal" but that was going to need a bigger refactor